### ISSUE TITLE
redirect_to parameter on Registration URL doesn't behave as expected

### ISSUE THREAD

**Issue by skillmatic-co** [ultimatemember/ultimatemember#665]
Adding `redirect_to` parameter to your Registration page url doesn't behave as expected. For example, if you had https://www.example.com/register?redirect_to=/after-registration, you would expect, after successful registration, for it to log you in, and then redirect to https://www.example.com/after-registration. This doesn't happen, instead, you get redirected to that URL and bypass auto login and profile slug generation, for example. You can see that in `um-actions-register.php` on lines 178-181:

```
// Priority redirect
if ( isset( $args['redirect_to'] ) ) {
    exit( wp_redirect( urldecode( $args['redirect_to'] ) ) );
}
```

I'm not sure why it checks for that parameter right off the bat and then doesn't execute any of the rest of the necessary functions.

Ideally, those lines would turn into:

```
// Priority redirect
if ( isset( $args['redirect_to'] ) ) {
    if ( $status == 'approved' ) {
        UM()->user()->auto_login( $user_id );
        UM()->user()->generate_profile_slug( $user_id );
    }
    exit( wp_redirect( urldecode( $args['redirect_to'] ) ) );
}
```

Or is there something I'm missing? This seems like a bug to me.

**Comment by OnlySid** [comment]
Having this exact same issue ^

**Comment by skillmatic-co** [comment]
> Having this exact same issue ^
> 
> Something similar is in um-actions-login.php preventing login redirect rules with user roles.

@OnlySid unfortunately I think this repo has been abandoned. There's not much activity here. You can view my PR request if you want, and make your own patch. I was able to fix this problem in my functions.php file though.

**Comment by OnlySid** [comment]
I’ll also submit a ticket to them; this made me hesitant to update to 2.1.7 but finally bit the bullet and did some patchwork where necessary. Hoping I didn’t screw anything else up in the process!

I’ll post here if I receive any replies from them.

**Comment by nikitasinelnikov** [comment]
Hi

This issue will be fixed in the next 2.1.8 version. It will be released as soon as possible this week.

Thanks!
